src/security/ServerOptions.cc

```diff
@@ -390,12 +396,12 @@ Security::ServerOptions::loadDhParams()
     parsedDhParams.resetWithoutLocking(dhp);
 
 #else // OpenSSL 3.0+
-    const auto type = eecdhCurve.isEmpty() ? "DH" : "EC";
+    const auto type = "DH";
 
     Ssl::ForgetErrors();
     EVP_PKEY *rawPkey = nullptr;
     using DecoderContext = std::unique_ptr<OSSL_DECODER_CTX, HardFun<void, OSSL_DECODER_CTX*, &OSSL_DECODER_CTX_free> >;
-    if (const DecoderContext dctx{OSSL_DECODER_CTX_new_for_pkey(&rawPkey, "PEM", nullptr, type, 0, nullptr, nullptr)}) {
+    if (const DecoderContext dctx{OSSL_DECODER_CTX_new_for_pkey(&rawPkey, "PEM", nullptr, type, OSSL_KEYMGMT_SELECT_DOMAIN_PARAMETERS, nullptr, nullptr)}) {
```

Comment:
Should we also allow so called "other" parameters (i.e. [OSSL_KEYMGMT_SELECT_OTHER_PARAMETERS](https://docs.openssl.org/3.0/man7/provider-keymgmt/#key-objects)) here? For example:

```suggestion
    if (const DecoderContext dctx{OSSL_DECODER_CTX_new_for_pkey(&rawPkey, "PEM", nullptr, type, OSSL_KEYMGMT_SELECT_ALL_PARAMETERS, nullptr, nullptr)}) {
```

I cannot tell whether those "other" parameters can be compatible with how Squid uses parsed/loaded parameters. If you are not sure, and tests pass with more permissive OSSL_KEYMGMT_SELECT_ALL_PARAMETERS, then perhaps we should err on the side of allowing more?